pkg/common/cns-lib/volume/manager.go

```diff
@@ -138,10 +138,18 @@ var (
 	volumeTaskMap       = make(map[string]*createVolumeTaskDetails)
 	// volumeTaskMapLock is used to serialize writes to volumeTaskMap.
 	volumeTaskMapLock sync.Mutex
+	snapshotTaskMap   = make(map[string]*createSnapshotTaskDetails)
```

Comment:
There's a https://pkg.go.dev/sync#Map, maybe it can solve the problem without creating explicit locks.

Reply:
Good point. But I will keep it as it is for the PR. It can be improved with Sync.Map for both volume and snapshot cases in a separate PR.